libs/awscdk/src/platform.ts

```diff
@@ -8,7 +36,62 @@ export class Platform implements platform.IPlatform {
   /** Platform model */
   public readonly target = "awscdk";
 
-  public newApp?(appProps: any): any {
+  public newApp(appProps: any): any {
     return new App(appProps);
   }
+
+  public newInstance(
```

Comment:
I think we should deprecate this API because these two APIs are competing with each other.

When will `newInstance` get called? When will `typeForFqn` get called?

If I only override the former, then base classes will not be injected.

I think we should just keep the latter.

Reply:
We discussed this a bit and both agreed that its probably safe to deprecate `newInstance()` but maybe do this in a followup PR. Just because there already a lot here, and this would require a decent amount of documentation updating as well.

Reply:
https://github.com/winglang/wing/issues/6871